### PR TITLE
Handle legacy browsers for TV

### PR DESCRIPTION
**Pull request by enjikaka**
This adds additional builds and entrypoints to the auth, event-producer and player modules. This is needed due to the oldest LG webOS we support is running on chrome76.

Esbuild - which Vite is using underneath - does not have transforms for top-level await, so a plugin is added for that.

This should enable the import of `@tidal-music/auth/legacy`, `@tidal-music/event-producer/legacy` and `@tidal-music/player/legacy`, but custom aliases in webpack also work.

### CODE REVIEW

File: packages/auth/package.json [hunk 1]
```diff
@@ -13,14 +13,13 @@
...
   "exports": {
-    ".": {
-      "import": "./dist/index.js",
-      "types": "./dist/index.d.ts",
-      "default": "./dist/index.js"
-    }
+    "import": "./dist/index.js",
+    "types": "./dist/index.d.ts",
+    "default": "./dist/index.js",
+    "legacy": "./dist/legacy/index.js"
```
Comment (MathiasTim) [inline packages/auth/package.json:19]: Was there any reason why we set it up with that dot? I always thought it's a bit weird and good if we can get rid off.
Maybe should update common, etc. as well then?

Comment (enjikaka) [reply packages/auth/package.json:19]: > Was there any reason why we set it up with that dot?

I just think it's a confusing copy paste, it's for scoping sub-parts and for sorting CJS/ESM. Some people on the internet does it and I suppose Haffy copy pasted some code from someone that did it. As I understand it's not supposed to be used for the imports, types, and default ones - and we also only create esm bundles. The Nodejs documentation on it: https://nodejs.org/api/packages.html#conditional-exports

> Maybe should update common, etc. as well then?

I'll do that!

Comment (osmestad) [reply packages/auth/package.json:19]: I guess we'd need to review the old git history before this was public to see? the `player` package does not have that though, so I assume it should work without :-)

File: packages/player/package.json [hunk 2]
```diff
@@ -5,15 +5,16 @@
   "type": "module",
   "exports": {
     "import": "./dist/index.js",
-    "default": "./dist/index.js"
+    "default": "./dist/index.js",
+    "legacy": "./dist/legacy/index.js"
```
Comment (osmestad) [inline packages/player/package.json:9]: Should we add the "types" under "exports" here as well? (to be consistent with the other packages)

File: packages/event-producer/package.json [hunk 3]
```diff
@@ -13,15 +13,14 @@
   "module": "dist/index.js",
   "types": "dist/index.d.ts",
   "exports": {
-    ".": {
-      "import": "./dist/index.js",
-      "types": "./dist/index.d.ts",
-      "default": "./dist/index.js"
-    }
+    "import": "./dist/index.js",
+    "types": "./dist/index.d.ts",
```
Comment (osmestad) [inline packages/event-producer/package.json:17]: One `types` left here I think?

Comment (osmestad) [inline packages/event-producer/package.json:17]:
```suggestion
```

**Review by osmestad — APPROVED** [review]

**Review by MathiasTim — APPROVED** [review]

**Pull request merged**